NeoML/include/NeoML/Dnn/Dnn.h

```diff
@@ -496,6 +499,25 @@ NEOML_API IMathEngine* GetRecommendedGpuMathEngine( size_t memoryLimit );
 
 //------------------------------------------------------------------------------------------------------------
 
+// Manages reference counts and initial states for CDnn objects, restoring configurations once all references are removed
+class NEOML_API CDnnReferenceRegistor final {
+public:
+	CDnnReferenceRegistor();
+	explicit CDnnReferenceRegistor(CDnn* _originalDnn);
+	CDnnReferenceRegistor& operator=(CDnnReferenceRegistor&& other);
+
+private:
+	~CDnnReferenceRegistor();
+
+	bool learningState = true; // Initial learning state of parent Dnn (before creating references)
+	int referenceCounter = 0; // -1 if reference, else - number of child dnns
```

Comment:
There are mistakes
`// For the reference dnn, the counter = -1, else for the original dnn it stores the number of reference dnns of this dnn`

Please, do not use `child`, use same terms everywhere, do not create additional synonyms